File: BitManipulation/counting_bits.py
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        bin_list = []
        for i in range(num+1):
            bin_list.append(bin(i)[2:].count('1'))
        return bin_list

File: BitManipulation/CountingBits.java
class Solution {
    public int[] countBits(int num) {
        
        int[] nums = new int[num+1];
        
        for (int i = 0; i <= num; i++) {
            
            int count = 0;
            
            for (int offset = 0; offset < 32; offset++) {
                if (((i >> offset) & 1) == 1) count++;
            }
            
            nums[i] = count;
            
        }
        
        return nums;
        
    }
}
